extensions/common/vault/vault-gcp/src/main/java/org/eclipse/edc/vault/gcp/GcpSecretManagerVault.java

```diff
@@ -0,0 +1,252 @@
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.secretmanager.v1.AccessSecretVersionResponse;
+import com.google.cloud.secretmanager.v1.ProjectName;
+import com.google.cloud.secretmanager.v1.Replication;
+import com.google.cloud.secretmanager.v1.Secret;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceSettings;
+import com.google.cloud.secretmanager.v1.SecretName;
+import com.google.cloud.secretmanager.v1.SecretPayload;
+import com.google.cloud.secretmanager.v1.SecretVersion;
+import com.google.cloud.secretmanager.v1.SecretVersionName;
+import com.google.protobuf.ByteString;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.result.Result;
+import org.eclipse.edc.spi.security.Vault;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+
+/**
+ * Vault extension implemented with GCP Secret Manager.
+ */
+public class GcpSecretManagerVault implements Vault {
+
+    private final Monitor monitor;
+    private final String project;
+    private final String region;
+    private final SecretManagerServiceClient secretManagerServiceClient;
+    private final Object vaultLock;
+
+    private static final String LATEST_VERSION_ALIAS = "latest"; // alias for the latest version of a Secret
+    private static final int MAX_KEY_LENGTH = 255; // maximum Secret Manager key length
+    private static final int HASH_LENGTH = 8; // Object.hashCode() returns int, which is 32 bit, in hex results in 8 char
+
+    /**
+     * Factory helper constructing Vault object with default GCP credentials.
+     *
+     * @param monitor monitor object for logging.
+     * @param project GCP project name.
+     * @param region replica location for secrects created by the vault.
+     * @return the created Vault object backed by Secret Manager.
+     * @throws IOException if the creation of the Vault in GCP fails.
+     */
+    public static GcpSecretManagerVault createWithDefaultSettings(Monitor monitor, String project, String region) throws IOException {
+        return new GcpSecretManagerVault(monitor, project, region, SecretManagerServiceClient.create());
+    }
+
+    /**
+     * Factory helper constructing Vault object with service account GCP credentials.
+     *
+     * @param monitor monitor object for logging.
+     * @param project GCP project name.
+     * @param region replica location for secrects created by the vault.
+     * @param filePath path of the JSON file with service account credentials.
+     * @return the created Vault object backed by Secret Manager.
+     * @throws IOException if the creation of the Vault in GCP fails.
+     */
+    public static GcpSecretManagerVault createWithServiceAccountCredentials(Monitor monitor, String project, String region, String filePath) throws IOException {
```

Comment:
instead of the `filePath`, it's better to pass in a `Stream`, that way the vault remains agnostic and is more versatile.

Reply:
To be fair I think that it's better to shrink the stream scope as much as possible, unless it's needed otherwise.
In any case, please close the stream accordingly (preferably using try-with-resources)

Reply:
Moved to stream with reduced scope and closed explicitly, we had a try catch for the function in place, I would suggest to keep it.